esmvaltool/diag_scripts/climate_patterns/sub_functions.py

```diff
@@ -0,0 +1,217 @@
+"""Script containing relevant sub-functions for driving scripts.
+
+Author
+------
+Gregory Munday (Met Office, UK)
+"""
+
+import logging
+import multiprocessing as mp
+import os
+from functools import partial
+from pathlib import Path
+
+import iris
+import iris.analysis.cartography
+import iris.coord_categorisation
+import numpy as np
+
+logger = logging.getLogger(Path(__file__).stem)
+
+
+def load_cube(filename):
+    """Load cube, remove any dimensions of length: 1.
+
+    Parameters
+    ----------
+    filename : path
+        path to load cube file
+
+    Returns
+    -------
+    cube : cube
+        a cube
+    """
+    logger.debug("Loading %s", filename)
+    cube = iris.load_cube(filename)
+    cube = iris.util.squeeze(cube)
+
+    return cube
+
+
+def ocean_fraction_calc(sftlf):
+    """Calculate gridded land and ocean fractions.
+
+    Parameters
+    ----------
+    sftlf: cube
+        land-fraction cube from piControl experiment
+
+    Returns
+    -------
+    ocean_frac: cube
+        ocean_fraction cube for area-weights
+    land_frac: cube
+        land_fraction cube for area-weights
+    of: float
+        ocean fraction float for EBM calculations
+    """
+    sftlf.coord("latitude").coord_system = iris.coord_systems.GeogCS(
+        6371229.0
+    )
+    sftlf.coord("longitude").coord_system = iris.coord_systems.GeogCS(
+        6371229.0
+    )
+    sftof = sftlf.copy()
+    sftof.data = 100.0 - sftlf.data
+
+    ocean_frac = sftof / 100
+    land_frac = sftlf / 100
+
+    return ocean_frac, land_frac
+
+
+def area_avg_landsea(cube, ocean_frac, land_frac, land=True, return_cube=None):
+    """Calculate the global mean of a variable in a cube.
+
+    Parameters
+    ----------
+    cube : cube
+        input cube
+    ocean_frac : cube
+        ocean fraction cube, found from sftlf
+    land_frac : cube
+        land fraction cube, sftlf
+    land : bool
+        option to weight be land or ocean
+    return_cube : bool
+        option to return a cube or array
+
+    Returns
+    -------
+    cube2 : cube
+        cube with collapsed lat-lons, global mean over time
+    cube2.data : arr
+        array with collapsed lat-lons, global mean over time
+    """
+    if not cube.coord("latitude").has_bounds():
+        cube.coord("latitude").guess_bounds()
+    if not cube.coord("longitude").has_bounds():
+        cube.coord("longitude").guess_bounds()
+
+    global_weights = iris.analysis.cartography.area_weights(
+        cube,
+        normalize=False
+    )
+
+    if land is False:
+        ocean_frac.data = np.ma.masked_less(ocean_frac.data, 0.01)
+        weights = iris.analysis.cartography.area_weights(
+            ocean_frac,
+            normalize=False
+        )
+        ocean_area = (
+            ocean_frac.collapsed(
+                ["latitude", "longitude"], iris.analysis.SUM, weights=weights
+            )
+            / 1e12
+        )
+        print("Ocean area: ", ocean_area.data)
```

Comment:
Would it be possible to use the logger here rather than `print`? 🤔

Reply:
I've actually just removed this entirely, as it's unnecessary